### PR TITLE
rgw: hiding parquet dependencies upon parquet-select option is off

### PR DESCRIPTION
**Pull request by galsalomon66**
upon a user wishes to build RGW without parquet (option is off), the parquet code-lines should be masked.   

fixes: https://tracker.ceph.com/issues/68829

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by mkogan1 — APPROVED** [review]
LGTM, 
this fixes the build error with `-DWITH_RADOSGW_SELECT_PARQUET=OFF`

File: src/rgw/rgw_s3select.cc [hunk 1]
```diff
@@ -762,7 +762,9 @@ void RGWSelectObj_ObjStore_S3::execute(optional_yield y)
       op_ret = -ERR_INVALID_REQUEST;
     } else {
       //status per amount of processed data
+//#ifdef _ARROW_EXIST
```
[the remark this answers is not in the document]
Comment (galsalomon66) [reply src/rgw/rgw_s3select.cc:765]: > why adding a comment? or is there a pre-processor manipulating this file before compilation?

it's a temporary. 
i ran teuthology, it failed, i want to verify it is not related to the recent change.

[the remark this answers is not in the document]
Comment (ronen-fr) [reply src/rgw/rgw_s3select.cc:765]: What I was trying to say: as this is an approved PR, it probably should not be merged with those comments

[the remark this answers is not in the document]
Comment (cbodley) [reply src/rgw/rgw_s3select.cc:765]: > i ran teuthology, it failed, i want to verify it is not related to the recent change.

@galsalomon66 the failures in https://pulpito.ceph.com/gsalomon-2024-11-06_12:03:18-rgw:verify-mask_parquet_dependencies-distro-default-smithi/ look like they're caused by the RGWPSListTopicsOp crashes tracked in https://tracker.ceph.com/issues/68756 (cc @yuvalif)

[the remark this answers is not in the document]
Comment (galsalomon66) [reply src/rgw/rgw_s3select.cc:765]: Thanks @cbodley

**Comment by jmundack** [comment]
fair to assume we can get this in once the PR checks complete?

**Comment by galsalomon66** [comment]
jenkins test api

**Comment by yuvalif** [comment]
jenkins test api

**Comment by galsalomon66** [comment]
https://github.com/ceph/ceph/pull/62959 (backport into squid)

**Pull request merged**